install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -318,11 +318,18 @@ spec:
       - name: cilium-monitor
         image: {{ include "cilium.image" .Values.image | quote }}
         imagePullPolicy: {{ .Values.image.pullPolicy }}
-        command: ["cilium"]
+        command:
+        - /bin/bash
+        - -c
+        - --
         args:
-        - monitor
-        {{- range $type := .Values.monitor.eventTypes }}
-        - --type={{ $type }}
+        - |-
+          for i in {1..5}; do \
+            [ -S /var/run/cilium/monitor1_2.sock ] && break || sleep 10;\
+            done; \
```

Comment:
startupProbe is commonly used for slow starting app, what do you think ?

Reply:
the problem is that, the cilium-agent starts fine but `cilium-monitor` container when socket doesn't preset it return exit code 1 and container with cilium monitor is restart. For example, retries inside `cilium monitor` create a new one flag -retry 10 or flag -delay 60s for example, then it fixed the issue
startupProbe is for delay before redinnesProbe/LivenessProbe run, so startupProbe won't help in this case :(

Reply:
example:
```
  cilium-monitor:
    Container ID:  containerd://150a7dfe567bcaefaeae2b11d09dac6be8b2c04f4a03937eb8f27e09925bae22
    Image:         quay.io/cilium/cilium:v1.11.1@sha256:251ff274acf22fd2067b29a31e9fda94253d2961c061577203621583d7e85bd2
    Image ID:      quay.io/cilium/cilium@sha256:251ff274acf22fd2067b29a31e9fda94253d2961c061577203621583d7e85bd2
    Port:          <none>
    Host Port:     <none>
    Command:
      cilium
    Args:
      monitor
      --type=drop
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Fri, 08 Apr 2022 16:29:12 +0300
      Finished:     Fri, 08 Apr 2022 16:29:12 +0300
    Ready:          False
    Restart Count:  2
    Limits:
      memory:  128Mi
    Requests:
      cpu:        10m
      memory:     64Mi
    Startup:      http-get http://127.0.0.1:9876/healthz delay=0s timeout=1s period=2s #success=1 #failure=105
    Environment:  <none>
    Mounts:
      /var/run/cilium from cilium-run (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-5sgwk (ro)
```
```
  Warning  BackOff    16s (x4 over 19s)  kubelet            Back-off restarting failed container
  Warning  Unhealthy  9s (x6 over 19s)   kubelet            Startup probe failed: Get "http://127.0.0.1:9876/healthz": dial tcp 127.0.0.1:9876: connect: connection refused
```
```
NAME           READY   STATUS             RESTARTS      AGE
cilium-zqzp5   0/2     CrashLoopBackOff   1 (11s ago)   15s
cilium-zqzp5   0/2     Completed          2 (14s ago)   18s
cilium-zqzp5   0/2     CrashLoopBackOff   2 (2s ago)    20s
cilium-zqzp5   0/2     Running            3 (24s ago)   42s
cilium-zqzp5   0/2     Running            3 (24s ago)   42s
cilium-zqzp5   0/2     Running            3 (24s ago)   42s
cilium-zqzp5   1/2     Running            3 (25s ago)   43s
cilium-zqzp5   2/2     Running            3 (25s ago)   43s
```
```
Press Ctrl-C to quit
time="2022-04-08T13:28:58Z" level=error msg="Cannot open monitor socket" error="Cannot find or open a supported node-monitor socket. /var/run/cilium/monitor1_2.sock: dial unix /var/run/cilium/monitor1_2.sock: connect: no such file or directory"
```

Reply:
If this solution is bad, then should be figure out what the flag add into `monitor` -> https://github.com/cilium/cilium/issues/19364 like addition delay or retries to check sock file

Reply:
or maybe, just add sleep before cilium command is run
for example 
```
sleep 60
cilium monitor
-t drop
```

Reply:
I think the current solution is the right approach.

Reply:
thanks for chiming in :1st_place_medal: